java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/ArrowFlightConnectionConfigImpl.java

```diff
@@ -137,7 +138,14 @@ public int threadPoolSize() {
    */
   public CallOption toCallOption() {
     final CallHeaders headers = new FlightCallHeaders();
-    properties.forEach((key, val) -> headers.insert(key.toString(), val.toString()));
+    ArrowFlightConnectionProperty[] builtInProperties = ArrowFlightConnectionProperty.values();
+    properties.forEach(
+        (key, val) -> {
+          // For built-in properties before adding new headers
+          if (Arrays.stream(builtInProperties).noneMatch(buildinPropertie -> buildinPropertie.camelName.equals(key.toString()))) {
```

Comment:
buildInPropertie should be renamed to builtInProperty.

The comparison used should be case insensitive

Reply:
done